### PR TITLE
[repo] Update dotnet format ci job

### LINKED ISSUE #4839: [repo] Replace outdated tool dotnet-format by the built-in dotnet format
Today we have a CI [workflow](https://github.com/open-telemetry/opentelemetry-dotnet/blob/459783f700b87325e057493d0715338d93e2de63/.github/workflows/dotnet-format.yml#L26-L30) which uses a .NET tool called `dotnet-format` to do (essentially) style checks on any code being added to the repo.

`dotnet-format` is now part of the .NET SDK and can be invoked as `dotnet format` from the CLI.

We want to switch our CI workflow to the new style because the old one doesn't work for some of the newer things like raw string literals.

I already did this in the contrib repo: https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/1339

Throwing this issue up in case anyone wants to take it on.

It should hopefully be simpler in the main repo because there aren't any legacy AspNet projects. I would suggest to start with updating the workflow with what contrib has (https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/1339/files#diff-ca8c2611c79b991c0fbe04fec3c97c14dc83419f5efb1e8a7a96dd51e7df3e2a) and go from there.

### PR DESCRIPTION
**Pull request by tombiddulph**
Updated `dotnet-format.yml` to use `dotnet format` from the dotnet sdk.

Formatted solution using `dotnet format`

Fixes #4839 
Design discussion issue #

## Changes
Replace outdated tool dotnet-format by the built-in dotnet format

## Merge requirement checklist

* [x] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/CONTRIBUTING.md) guidelines followed (nullable enabled, static analysis, etc.)
* [ ] Unit tests added/updated
* [ ] Appropriate `CHANGELOG.md` files updated for non-trivial changes
* [ ] Changes in public API reviewed (if applicable)

### CODE REVIEW

**Review by pellared — APPROVED** [review]

**Comment by Kielek** [comment]
@tombiddulph, please check issues detected by https://github.com/open-telemetry/opentelemetry-dotnet/actions/runs/6290937854/job/17092815581?pr=4881.

**Comment by tombiddulph** [comment]
@Kielek I dont' seem to get the same errrors when running locally. I've got a gist [here](https://gist.github.com/tombiddulph/a856efa04a86a447484a00e45f0b8ed2) of the output when running locally. Note that I'm running this on MacOS. I'll give it a go again on Windows a bit later to see if I get the same result.

File: src/OpenTelemetry.Api.ProviderBuilderExtensions/Logs/OpenTelemetryDependencyInjectionLoggerProviderBuilderExtensions.cs [hunk 1]
```diff
@@ -27,11 +27,11 @@ namespace OpenTelemetry.Logs;
 /// Contains extension methods for the <see cref="LoggerProviderBuilder"/> class.
 /// </summary>
 #if EXPOSE_EXPERIMENTAL_FEATURES
-    public
+public
 #else
     internal
 #endif
```
Comment (reyang) [inline src/OpenTelemetry.Api.ProviderBuilderExtensions/Logs/OpenTelemetryDependencyInjectionLoggerProviderBuilderExtensions.cs:33]: This looks weird (no indentation on `public`, with indentation on `internal`) - is this a bug in the tool?

Comment (tombiddulph) [reply src/OpenTelemetry.Api.ProviderBuilderExtensions/Logs/OpenTelemetryDependencyInjectionLoggerProviderBuilderExtensions.cs:33]: It looks like code hidden within preprocessor directives in not formatted by dotnet format see https://github.com/dotnet/format/issues/783, I'd guess that the code exposed by the preprocessor directives isn't formatted properly due to directives breaking up definitions although I haven't looked into this to confirm it.

**Comment by Kielek** [comment]
> @Kielek I don't seem to get the same errrors when running locally. I've got a gist [here](https://gist.github.com/tombiddulph/a856efa04a86a447484a00e45f0b8ed2) of the output when running locally. Note that I'm running this on MacOS. I'll give it a go again on Windows a bit later to see if I get the same result.

I suppose that you have already restored packages locally.
Consider to call `git clean -fxd` and then call dotnet format.

**Comment by tombiddulph** [comment]
Yep I think that was the issue, the packages were restored locally however the first commit to `dotnet-formal.yml` included the `--no-restore` flag.

**Review by Kielek — APPROVED** [review]

File: src/OpenTelemetry.Api.ProviderBuilderExtensions/Logs/OpenTelemetryDependencyInjectionLoggerProviderBuilderExtensions.cs (hunk 1 above)
Comment (utpilla) [reply src/OpenTelemetry.Api.ProviderBuilderExtensions/Logs/OpenTelemetryDependencyInjectionLoggerProviderBuilderExtensions.cs:33]: Stay consistent with the indentation. Either keep it for both `public` and `internal` or remove it for both.

**Comment by tombiddulph** [comment]
Commenting to keep open

**Comment by utpilla** [comment]
> Commenting to keep open

@tombiddulph Could you address this comment? https://github.com/open-telemetry/opentelemetry-dotnet/pull/4881#discussion_r1351207311

**Review by CodeBlanch — APPROVED** [review]
LGTM!

I'm going to merge this and then do a follow-up so we run format in stable & experimental modes. I think that will force the one styling issue brought up to be consistent

**Pull request merged**